src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,73 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.mapper.ProductMapper;
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+
+    public ProductController(ProductService productService,
+                             ProductMapper productMapper) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        Product product = productService.save(productMapper.mapToModel(requestDto));
+        return productMapper.mapToDto(product);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto requestDto) {
+        Product product = productMapper.mapToModel(requestDto);
+        product.setId(id);
+        productService.update(product);
+        return productMapper.mapToDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.mapToDto(productService.getById(id));
+    }
+
+    @DeleteMapping({"/id"})
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @GetMapping("/by-price")
+    public List<ProductResponseDto> getAllByPriceBetween(@RequestParam BigDecimal from,

```

Comment:
```suggestion
    public List<ProductResponseDto> findAllByPriceBetween(@RequestParam BigDecimal from,
```
Keep to consistency with service layer method